NMS-9684: Introduce version compatibility list for Horizon 25.1.0

indigo423 opened the pull request:
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

Add a version compatibility list in the Admin guide which gives administrators the information with which components this specific version of OpenNMS Horizon is compatible with.

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-9684

RangerRick commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,18 @@
+| Newts                   | 1.5.2
+| Helm                    | 3+
+| Kafka                   | 2.3.0
+| Elasticsearch           | 6.2.4 until 7.3.0
```
Pretty sure we don't support Elasticsearch 6.x in Horizon 25.

indigo423 replied:
What I know here so far, Elasticsearch version dependency is driven by our elasticsearch-drift-plugin which is installed on the Elasticsearch node. The first version was 1.0.0 which pointed to Elasticsearch 6.2.4 which is the reason I've picked it :) Now is the best chance to clarify this :)

j-white replied:
H25 -> Only ES 7.x: http://docs.opennms.org/opennms/releases/25.0.0/releasenotes/releasenotes.html#_elasticsearch_7_x_support

indigo423 commented:
> Also this should be targeted to `foundation-2019` rather than `develop`.

Done! Thanks good point :)

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,18 @@
+| Cassandra               | 3.11+
+| Newts                   | 1.5.2
+| Helm                    | 3+
+| Kafka                   | 2.3.0
```
That's the version of the client library we use, but it's backwards compatible.

indigo423 replied:
@j-white Yes, I've picked it up from the pom.xml. Should we write 1.x - 2.x here?

j-white replied:
Works for me.

RangerRick commented:
I haven't really done much with it, but as I understand it, besides the plugin there is also code and schemas on the OpenNMS side that require Eliasticsearch 7 because it's difficult to support both simultaneously.  @j-white or @mattixtech could probably answer better.

RangerRick commented:
Oops, for some reason I didn't get an email about Jesse's comment, but yeah, ES7 only.  :)

indigo423 commented:
Do we follow the [semantic version](https://semver.org/) for our own components? If so, we could write here for our own components `<major version number>.x`, i.e. Helm, Newts, OIA? What about ALEC? Can we break compatibility with new ALEC versions which aren't compatible with Horizon Versions?

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,18 @@
+|===
+| Component               | Version Compatibility
+| OpenNMS Helm            | 3+
+| OpenNMS Integration API | 0.2.1
```
```suggestion
| OpenNMS Integration API | 0.2.x
```

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,18 @@
+| Component               | Version Compatibility
+| OpenNMS Helm            | 3+
+| OpenNMS Integration API | 0.2.1
+| Newts                   | 1.5.2
```
Given that Newts is embedded, do we need to mention it here?

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,18 @@
+| OpenNMS Integration API | 0.2.1
+| Newts                   | 1.5.2
+| Cassandra               | 3.11.+
+| Elasticsearch           | 7.x - 7.3.0
```
```suggestion
| Elasticsearch           | 7.x
```

indigo423 commented:
Thank you for help.

indigo423 commented:
@j-white @RangerRick anything we need to add here?

RangerRick commented on opennms-doc/guide-admin/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,17 @@
+| Elasticsearch           | 7.x
+| Java Development Kit    | OpenJDK 8, OpenJDK 11
+| Kafka                   | 2.3.0
+| PostgreSQL              | 10.x - 11.x
```
we technically allow 10-12, only recommend 11 because 12 had come out moments before the horizon release

indigo423 commented:
@j-white @RangerRick I've fixed all comments related compatibility. I noticed two things:

1) I've moved it from the admin guide to the install guide, because this is the place where you need the information, what you need to install things. Any objections here?
2) What do you think about adding supported operating systems to the compatibility list?

Thank you in advance.

j-white commented on opennms-doc/guide-install/src/asciidoc/text/compatibility.adoc:
```diff
@@ -0,0 +1,17 @@
+| Cassandra               | 3.11.+
+| Elasticsearch           | 7.x
+| Java Development Kit    | OpenJDK 8, OpenJDK 11
+| Kafka                   | 2.3.0
```
```suggestion
| Kafka                   | 1.x - 2.x
```

j-white commented:
Looks good - just the Kafka version needs updating.

RangerRick commented:
>     1. I've moved it from the admin guide to the install guide, because this is the place where you need the information, what you need to install things. Any objections here?

Makes sense.

>     2. What do you think about adding supported operating systems to the compatibility list?

Couldn't hurt I guess, though since it's JVM-based there isn't that much that _isn't_ technically compatible, especially if you use all JNA-based stuff.  Not sure it's worth trying to nail down specifics.

indigo423 commented:
> Looks good - just the Kafka version needs updating.

Fixed, sorry I haven't understood the conversation in the comments correctly :)

indigo423 commented:
> > ```
> > 1. I've moved it from the admin guide to the install guide, because this is the place where you need the information, what you need to install things. Any objections here?
> > ```
> 
> Makes sense.
> 
> > ```
> > 2. What do you think about adding supported operating systems to the compatibility list?
> > ```
> 
> Couldn't hurt I guess, though since it's JVM-based there isn't that much that _isn't_ technically compatible, especially if you use all JNA-based stuff. Not sure it's worth trying to nail down specifics.

Ok we can skip this in the first place and think about it more, adding stuff later isn't a big deal.
